apiutils/task.go

```diff
@@ -0,0 +1,21 @@
+package apiutils
+
+import (
+	"reflect"
+
+	"github.com/docker/swarmkit/api"
+)
+
+// TasksSpecEqual returns true if the tasks are functionaly equal, ignoring status,
+// version and other superfluous fields.
+//
+// This used to decide whether or not to propagate a task update to a controller.
+func TasksSpecEqual(a, b *api.Task) bool {
```

Comment:
This function name is quite wrong. It includes the entire task, not just the spec.